### PR TITLE
Add support to run integration tests in macOS

### PR DESCRIPTION
**Pull request by tonyredondo**
@DataDog/apm-dotnet

### CODE REVIEW

File: test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs [hunk 1]
```diff
@@ -424,6 +437,16 @@ private string GetProfilerProjectBin()
...
+        private string GetProfilerProjectNonWindowsBin()
+        {
+            return Path.Combine(
+                GetSolutionDirectory(),
+                "src",
+                "Datadog.Trace.ClrProfiler.Native",
+                "build",
+                "bin");
+        }
+
```
Comment (andrewlock) [inline test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs:449]: Is it possible to make the profiler build to the same dir in both Windows and Non-Windows? I feel like this discrepancy is going to cause other issues when we try and improve the build pipeline in general 🙁

Comment (tonyredondo) [reply test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs:449]: Ok, no problem, right now we simulate the windows folder structure: https://github.com/DataDog/dd-trace-dotnet/blob/tony/macos-integration-tests-fix/build/docker/Datadog.Trace.ClrProfiler.Native.sh#L15-L16

Comment (tonyredondo) [reply test/Datadog.Trace.TestHelpers/EnvironmentHelper.cs]: done in dd94206

**Review by lucaspimentel — APPROVED** [review]

**Review by zacharycmontoya — APPROVED** [review]

**Comment by tonyredondo** [comment]
Superseed by https://github.com/DataDog/dd-trace-dotnet/pull/1256

Closing it.

**Pull request closed without merging**